autopilot/contracts.go

```diff
@@ -1,27 +1,413 @@
 package autopilot
 
 import (
+	"math"
+	"time"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.sia.tech/renterd/bus"
 	"go.sia.tech/renterd/internal/consensus"
+	rhpv2 "go.sia.tech/renterd/rhp/v2"
 	"go.sia.tech/renterd/worker"
 	"go.sia.tech/siad/types"
 	"golang.org/x/crypto/blake2b"
 )
 
-func (ap *Autopilot) currentHeight() uint64 {
-	return ap.store.Tip().Height
+const (
+	// contractSetName defines the name of the default contract set
+	contractSetName = "autopilot"
+
+	// contractLoopInterval defines the interval with which we run the contract loop
+	contractLoopInterval = 15 * time.Minute
+
+	// estimatedFileContractTransactionSetSize is the estimated blockchain size
+	// of a transaction set between a renter and a host that contains a file
+	// contract.
+	estimatedFileContractTransactionSetSize = 2048
+)
+
+func logErr(err error) {} // TODO
+
+func (ap *Autopilot) contractLoop() {
+	ticker := time.NewTicker(contractLoopInterval)
+
+	for {
+		select {
+		// TODO: use build var for loop interval
+		// TODO: add wakeChan (to trigger loop)
+		case <-ap.stopChan:
+			return
+		case <-ticker.C:
+		}
+
+		// don't run the contract loop if we are not synced
+		if !ap.store.State().Synced {
+			continue
+		}
+
+		// re-use same state and config in every iteration
+		config := ap.store.Config()
+		state := ap.store.State()
+
+		// return early if no hosts are requested
+		if config.Contracts.Hosts == 0 {
+			return
+		}
+
+		// fetch our wallet address
+		address, err := ap.bus.WalletAddress()
```

Comment:
Why do this? I don't think we want to reuse addresses to not make it super obvious that a batch of contracts belongs to the same renter.

Also the comment makes it seem like a wallet only has a single address?

Reply:
correct, the plan is to use a single-address wallet, at least initially. Better UX for the vast majority of users.